config/routes.rb

```diff
@@ -1,4 +1,14 @@
+class AuthConstraint
+  def matches?(request)
+    request.session[:user]["email"].present?
```

Comment:
Oh awesome I didn't know that! Ok I will just do a check against the database with the email.